src/read.c

```diff
@@ -852,24 +852,40 @@ static avifResult avifDecoderItemMaxExtent(const avifDecoderItem * item, avifExt
 
     // construction_method: file(0)
 
+    if (sample->size == 0) {
+        return AVIF_RESULT_TRUNCATED_DATA;
+    }
+    size_t remainingBytes = sample->size; // This may be smaller than item->size if the item is progressive
+
     // Assert that the for loop below will execute at least one iteration.
     assert(item->extents.count != 0);
     uint64_t minOffset = UINT64_MAX;
     uint64_t maxOffset = 0;
     for (uint32_t extentIter = 0; extentIter < item->extents.count; ++extentIter) {
         avifExtent * extent = &item->extents.extent[extentIter];
+        const size_t usedExtentSize = (extent->size > remainingBytes) ? extent->size : remainingBytes;
```

Comment:
IMPORTANT: The ? : expression seems wrong. I think you intend to take the minimum, i.e.,

`const size_t usedExtentSize = (extent->size > remainingBytes) ? remainingBytes : extent->size;`

Reply:
Oh, that does look backwards. Fixing.